### ISSUE TITLE
Use contextualised name on chooseAction(empty)

### ISSUE THREAD

**Issue by japgolly** [japgolly/test-state#37]
```scala
    dsl
      .chooseAction(NameFn {
        case None      => "A"
        case Some(ros) => "B"
      }) { ros =>
        dsl.emptyAction
      }
      .when(_ => true)
```

```scala
    dsl
      .chooseAction(NameFn {
        case None      => "A"
        case Some(ros) => "B"
      }) { ros =>
        dsl.action("C")(_ => ())
      }
      .when(_ => true)
```

should (on success) print "B" and "C" respectively. Instead it prints "A" and "C"

**Comment by japgolly** [comment]
maybe have an option to handle empty differently? sometimes remove, sometimes skip...?

**Comment by japgolly** [comment]
Fixed via 69527cbd422b5cd04b3a4fc6d737fd73c49c37dd
